plugins/scaffolder-common/src/api.ts

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { JsonObject, JsonValue, Observable } from '@backstage/types';
+import { JSONSchema7 } from 'json-schema';
+import { TaskSpec, TaskStep } from './TaskSpec';
+import type {
+  TemplateEntityV1beta3,
+  TemplateParameterSchema,
+} from './TemplateEntityV1beta3';
+import { TaskStatus } from '../client/src/schema/openapi';
+
+/**
+ * Options you can pass into a Scaffolder request for additional information.
+ *
+ * @public
+ */
+export interface ScaffolderRequestOptions {
+  token?: string;
+}
+
+/**
+ * The status of each task in a Scaffolder Job
+ *
+ * @public
+ */
+export type ScaffolderTaskStatus =
+  | 'cancelled'
+  | 'completed'
+  | 'failed'
+  | 'open'
+  | 'processing'
+  | 'skipped';
+
+/**
+ * The shape of each task returned from the `scaffolder-backend`
+ *
+ * @public
+ */
+export type ScaffolderTask = {
+  id: string;
+  spec: TaskSpec;
+  status: ScaffolderTaskStatus;
+  lastHeartbeatAt?: string;
+  createdAt: string;
+};
+
+/**
+ * A single action example
+ *
+ * @public
+ */
+export type ActionExample = {
+  description: string;
+  example: string;
+};
```

Comment:
Are we able to pull these from the spec as well?

Reply:
These two specifically? Or other API types in [`plugins/scaffolder-common/src/api.ts`](https://github.com/backstage/backstage/pull/27771/files/233b9e97f969c84194c1707f2d037f04b2d2657c#diff-1b5ad46439cb89690c8103b84f339965bc7fbb60b08f33f66e347ebe1e78b26f) as well?

Reply:
The other types in this file as well :)

Reply:
Sounds good 👍🏼 For types in `scaffolder-node`, such as [`TaskStatus`](https://github.com/backstage/backstage/blob/233b9e97f969c84194c1707f2d037f04b2d2657c/plugins/scaffolder-node/src/tasks/types.ts#L35) and [`SerializedTask`](https://github.com/backstage/backstage/blob/233b9e97f969c84194c1707f2d037f04b2d2657c/plugins/scaffolder-node/src/tasks/types.ts#L55), should we keep those defined there? Or also pull them from the spec through `scaffolder-common`?

Reply:
@aramissennyeydd - I couldn't pull beyond `ActionExample` and `ScaffolderTaskStatus` because most of the other types are using a little more "involved" types. For example, `ScaffolderTask` uses `TaskSpec` (which we're excluding from the spec), `Action` uses `JSONSchema7`, `ScaffolderScaffoldOptions` uses `JsonValue`, etc.
I'm happy to still move forward with pulling these types from the spec. Just wanted to confirm our direction.

Reply:
Let's keep the more complex ones here, any of the simple types would be great to pull directly from the spec though :)